src/features/content-explorer/content-explorer/ContentExplorer.js

```diff
@@ -508,12 +692,25 @@ class ContentExplorer extends Component {
                 >
                     {headerActionsAccessory}
                 </ContentExplorerHeaderActions>
-                {isSelectAllAllowed && (
-                    <ContentExplorerSelectAll
-                        numTotalItems={numTotalItems}
-                        isSelectAllChecked={isSelectAllChecked}
+                {isSelectAllAllowed && isIncludeSubfoldersAllowed ? (
```

Comment:
does this mean the new feature can only be used with `isSelectAllAllowed`? could we make them independent of each other?

something like:
```
<div>
    {isIncludeSubfoldersAllowed && <ContentExplorerIncludeSubfolders [...] />}
    {isSelectAllAllowed && <ContentExplorerSelectAll [...] />}
</div>
```
with `ContentExplorerSelectAll` taking a prop to optionally hide the label

this reuses the checkbox without having to add a separate one into `ContentExplorerIncludeSubfolders`

Reply:
As of right now, I believe the new feature can only be used with isSelectAllAllowed. This could potentially change in the future though

Reply:
that would be for the enterprise-reports use case though right? that could be handled by passing the appropriate props from the parent

I think it would be worth making them independent. changing it in the future might be a breaking change for other apps that start using this feature